sound/soc/sof/sof-pci-dev.c

```diff
@@ -225,7 +225,11 @@ EXPORT_SYMBOL_NS(sof_pci_remove, SND_SOC_SOF_PCI_DEV);
 
 void sof_pci_shutdown(struct pci_dev *pci)
 {
-	snd_sof_device_shutdown(&pci->dev);
+	struct snd_sof_dev *sdev = dev_get_drvdata(&pci->dev);
+
+	/* audio need to enter D3 before shutdown */
+	sdev->system_suspend_target = SOF_SUSPEND_S3;
+	snd_sof_suspend(&pci->dev);
```

Comment:
this has to be platform-specific. We cant apply this blanket rule for all platforms. What platforms have this requirement btw?

Reply:
and also the commit message and the target state are inconsistent. You talk about S5 in commit message and set the target to S3? Out of curiosity, why isnt hda_dsp_remove() like what we do in the shutdown callback for TGL not enough?

Reply:
hda_dsp_remove() is not enough based on IoTG's dump. Generally, they asked all devices are in the D3 state when the system state is in S5. From the the debug, they found there are still some registers not meeting the requirement. Maybe we can simply the sequence if customers complain the latency is too long.